addons/l10n_in/models/account_move_reversal.py

```diff
@@ -0,0 +1,14 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+
+class AccountMoveReversal(models.TransientModel):
+    _inherit = "account.move.reversal"
+
+    def reverse_moves(self, is_modify=False):
+        res = super().reverse_moves(is_modify)
+        if self.env.company.account_fiscal_country_id.code == "IN":
+            for move_line, new_move_line in zip(self.move_ids.line_ids, self.new_move_ids.line_ids):
```

Comment:
Can we be sure that there is always the same number of lines per move?
For instance, we could imagine that refunding an invoice would not create multiple payment terms but refund all at once.

Doing this on `invoice_line_ids` seems already like a safer approach.